Sources/main.swift

```diff
@@ -32,34 +31,63 @@ guard
 guard let databaseName = config["mysql", "database"]?.string else { throw BotError.missingMySQLDatabaseName }
 
 let mysql = try MySQL.Database(
-    host: host,
+    hostname: host,
     user: user,
     password: pass,
     database: databaseName
-)
+).makeConnection()
 
 // WebSocket Init
-let rtmResponse = try BasicClient.loadRealtimeApi(token: token)
+let rtmResponse = try loadRealtimeApi(token: token)
 
 guard let validChannels = rtmResponse.data["channels", "id"]?.array?.flatMap({ $0.string }) else { throw BotError.unableToLoadChannels }
 
 guard let webSocketURL = rtmResponse.data["url"]?.string else { throw BotError.invalidResponse }
 
-try WebSocket.connect(to: webSocketURL) { ws in
+func credit(_ ws: WebSocket, _ user: String, channel: String, threadTs: String?, printError: Bool = true) throws {
+    if true {
+        let total = try mysql.addCoins(for: user)
+        let response = SlackMessage(
+            to: channel,
+            text: "<@\(user)> has \(total) :coin:",
+            threadTs: threadTs
+        )
+        try ws.send(response)
+    }
+}
+
+try EngineClient.factory.socket.connect(to: webSocketURL) { ws in
     print("Connected ...")
 
     ws.onText = { ws, text in
         let event = try JSON(bytes: text.utf8.array)
         let last3Seconds = NSDate().timeIntervalSince1970 - 3
+        
+        let threadTs = event["thread_ts"]?.string
+
+        // reactions
+        if
```

Comment:
Restrict to valid channels and duplicate reactions

Reply:
@LoganWright can you explain what you need, i will try to build it, so we can get this pushed :) .. This seem to fix Penny locally at least :)